packages/cache/src/internal/tar.ts

```diff
@@ -1,155 +1,296 @@
 import {exec} from '@actions/exec'
+import {exportVariable} from '@actions/core'
 import * as io from '@actions/io'
 import {existsSync, writeFileSync} from 'fs'
 import * as path from 'path'
 import * as utils from './cacheUtils'
-import {CompressionMethod} from './constants'
+import {ArchiveTool} from './contracts'
+import {
+  CompressionMethod,
+  SystemTarPathOnWindows,
+  ArchiveToolType,
+  TarFilename,
+  ManifestFilename
+} from './constants'
 
 const IS_WINDOWS = process.platform === 'win32'
+exportVariable('MSYS', 'winsymlinks:nativestrict')
```

Comment:
This is a grave mistake. This code is supposed to be a library and not leave any remnants in the environment after it ran.

This can cause failures in GitHub workflow runs (and indeed, it did over here). You basically changed the behavior of every workflow run that involves MSYS2 (or for that matter, Git for Windows).

Reply:
Hi @dscho, I hope this was addressed in this [PR](https://github.com/actions/toolkit/pull/1329)?

Reply:
You are correct. We were in the process of getting it merged, will be creating a release to get it live.

Reply:
Thank you.